connect/http-client/src/main/java/org/camunda/connect/httpclient/impl/AbstractHttpConnector.java

```diff
@@ -69,15 +70,33 @@ public void setHttpClient(CloseableHttpClient httpClient) {
   @Override
   public R execute(Q request) {
     HttpRequestBase httpRequest = createHttpRequest(request);
-
     HttpRequestInvocation invocation = new HttpRequestInvocation(httpRequest, request, requestInterceptors, httpClient);
-
     try {
-      return createResponse((CloseableHttpResponse) invocation.proceed());
-    } catch (Exception e) {
-      throw LOG.unableToExecuteRequest(e);
-    }
+	  
+      R invocationResult = createResponse((CloseableHttpResponse) invocation.proceed());
+	  handleErrorResponse(request , invocationResult);
+	  return invocationResult;
+			
+	   } catch (Exception e) {
+	    throw LOG.unableToExecuteRequest(e);
+	  }
+  }
 
+ private void handleErrorResponse(Q request,R invocationResult ) {
+     Map<String, Object> configOptions = request.getConfigOptions();	
+	   
+     if( configOptions != null && invocationResult != null)  {
+	
+	    int statusCode = invocationResult.getStatusCode();
+	    String connectorResponse = invocationResult.getResponse();
+	    Optional<Object> handleHttpError = Optional.ofNullable(configOptions.get("throw-http-error"));  
+			
+	    if (handleHttpError.isPresent() && handleHttpError.get().toString()
+	        .equalsIgnoreCase("TRUE") && statusCode >= 400 && statusCode <= 599) {  
```

Comment:
🔧 Maybe a bit more simple with `orElse()` and `Boolean.parseBoolean`:

```suggestion
      Object handleHttpError = Optional.ofNullable(configOptions.get("throw-http-error")).orElse("FALSE");

      if (Boolean.parseBoolean(handleHttpError.toString()) && statusCode >= 400 && statusCode <= 599) {
```